src/nccl_ofi_rdma.c

```diff
@@ -3606,9 +3608,14 @@ static int recv(nccl_net_ofi_recv_comm_t *recv_comm, int n, void **buffers,
 	nccl_net_ofi_rdma_mr_handle_t **mr_handles = (nccl_net_ofi_rdma_mr_handle_t **)mhandles;
 	uint16_t msg_seq_num = 0;
 	bool eager = false;
+	bool recv_completion_optional = false;
 
 	assert(r_comm != NULL);
 
+	if (r_comm->base.base.dev_id && data_progress_auto && *base_req == (void *)NCCL_NET_OPTIONAL_RECV_COMPLETION) {
```

Comment:
why are you checking `dev_id` here?  Also agree with Raghu's previous comment, this should be checking for an rdma transport-level flag on whether we're using early completion.  That flag should be modifiable based on an env var.

Reply:
We should error if the user asks for early completion explicitly and we don't have progress auto.